docs/source/daml/intro/13_Interfaces.rst

```diff
@@ -0,0 +1,249 @@
+
+.. hint::
+  It is not mandatory to prefix interface names with the letter ``I``, but it
+  can be convenient to tell at a glance whether or not a type is an interface.
+
+.. literalinclude:: daml/daml-intro-13/daml/IAsset.daml
+  :language: daml
+  :start-after: -- INTERFACE_IASSET_DEF_BEGIN
+  :end-before: -- INTERFACE_IASSET_DEF_END
+
+.. literalinclude:: daml/daml-intro-13/daml/IAsset.daml
+  :language: daml
+  :start-after: -- INTERFACE_IASSET_TF_DEF_BEGIN
+  :end-before: -- INTERFACE_IASSET_TF_DEF_END
+
+There are a few things happening here:
+
+1. For each interface, we have defined a ``viewtype``. This is mandatory for all
+   interfaces. All viewtypes must be serializable records, since <insert
+   justification here>. This declaration means that the special ``view`` method,
+   when applied to a value of this interface, will return the specified type, in
+   this case ``VAsset``. This is the definition of ``VAsset``:
+
+   .. literalinclude:: daml/daml-intro-13/daml/IAsset.daml
+     :language: daml
+     :start-after: -- DATA_VASSET_DEF_BEGIN
+     :end-before: -- DATA_VASSET_DEF_END
+
+
+2. We have defined the methods ``setOwner`` and ``toTransferProposal`` as part
+   of the ``IAsset`` interface, and method ``asset`` as part of the
+   ``IAssetTransferProposal`` interface. Later, when we provide instances of
+   these interfaces, we'll see that it is mandatory to implement each of these
+   methods.
+
+3. We have defined the choice ``ProposeIAssetTransfer`` as part of the
+   ``IAsset`` interface, and the choices ``AcceptIAssetTransferProposal``,
+   ``RejectIAssetTransferProposal`` and ``WithdrawIAssetTransferProposal`` as
+   part of the ``IAssetTransferProposal`` interface. These correspond one-to-one
+   with the choices of ``Cash`` / ``CashTransferProposal`` and ``NFT`` /
+   ``NFTTransferProposal``.
+
+   You should notice that the choice controller and the choice body are defined
+   in terms of the methods that we bundled with the interfaces, including the
+   special ``view`` method. For example, the controller of
+   ``choice ProposeIAssetTransfer`` is ``(view this).owner``, that is, it's the
+   ``owner`` field of the ``view`` for the implicit current contract ``this``,
+   in other words, the owner of the current contract. The body of this choice is
+   ``create (toTransferProposal this newOwner)``, so it creates a new contract
+   whose contents are the result of applying the ``toTransferProposal`` method
+   to the current contract and the ``newOwner`` field of the choice argument.
+
+.. hint::
+  For a detailed explanation of the syntax used here, check out :ref:`daml-ref-interfaces`
+
+
+Interface instances
+-------------------
+
+On its own, an interface isn't very useful, since all contracts on the ledger
+must belong to some template type. In order to make the link between an
+interface and a template, we define an interface instance on either the
+template or the interface. In this example, we add
```

Comment:
Do we want to clarify what we mean here, maybe "by writing the instance inside either the interface body or the template body" - maybe would be good to amend `Cash.daml` to have one instance inside the interface too

Then again this might just confuse them

Reply:
reworded to "we define an interface instance inside the body of either the template or the interface"